src/SonarScanner.MSBuild.Tasks/Targets/SonarQube.Integration.targets

```diff
@@ -192,28 +196,46 @@
     <PropertyGroup Condition="$(IsTempProject) == 'true' AND $(SonarQubeExclude)==''" >
       <SonarQubeExclude>true</SonarQubeExclude>
     </PropertyGroup>
+    <Message Condition="$(IsTempProject) == 'true' AND $(SonarQubeExclude)==''" Importance="normal" Text="Sonar: ($(MSBuildProjectFile)) project is a temporary project and will be excluded" />
 
     <PropertyGroup Condition=" $(SonarQubeTestProject) == '' ">
       <!-- The MS Test project type guid-->
       <SonarQubeMsTestProjectTypeGuid>3AC096D0-A1C2-E12C-1390-A8335801FDAB</SonarQubeMsTestProjectTypeGuid>
-      <SonarQubeTestProject Condition=" $(ProjectTypeGuids.ToUpperInvariant().Contains('$(SonarQubeMsTestProjectTypeGuid)')) ">true</SonarQubeTestProject>
```

Comment:
Previously we just had the condition being checked in the `<SonarQubeTestProject>` property directly. However, we since we now want to write a message as well, I'm now evaluating the condition separately and using it to set a temporary variable in each case. 
The `<SonarQubeTestProject>` and `<Message>` elements both have simple conditions that refer to the temporary variable.